cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/service/InMemoryBusinessMetadataStore.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.data2.metadata.service;
+
+import co.cask.cdap.data2.metadata.dataset.BusinessMetadataRecord;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.metadata.MetadataRecord;
+import co.cask.cdap.proto.metadata.MetadataSearchTargetType;
+
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Implementation of {@link BusinessMetadataStore} used in memory mode.
+ */
+public class InMemoryBusinessMetadataStore implements BusinessMetadataStore {
```

Comment:
To make it clear that this metadata store does not do anything, it would be good to call it `NoOpBusinessMetadataStore`.